x/leverage/keeper/msg_server.go

```diff
@@ -376,6 +376,14 @@ func (s msgServer) MaxBorrow(
 	if err != nil {
 		return nil, err
 	}
+
+	// MeToken module supply is fully protected in order to guarantee its availability for redemption.
+	meTokenSupply, err := s.keeper.GetSupplied(ctx, s.keeper.meTokenAddr, msg.Denom)
+	if err != nil {
+		return nil, err
+	}
+	moduleMaxBorrow = sdk.MaxInt(moduleMaxBorrow.Sub(meTokenSupply.Amount), sdk.ZeroInt())
+
```

Comment:
maybe instead of adding this block, modify the `moduleMaxBorrow` function with this logic